### DISCUSSION TITLE
got error: "/plotly.js (404: Not Found)" when previewing a dashboard

### DISCUSSION THREAD

**Discussion by hopezh** [quarto-dev/quarto-cli#8833] [Q&A] [answered]
### Description

I got the following error while `quarto preview` the `qmd` file. 

```
Terminating existing preview server....DONE
Preparing to preview
Watching files for changes
Browse at http://localhost:5242/
GET: /
  /plotly.js (404: Not Found)

```
May I ask if I missed anything important here? 

My quarto=v1.4.550, plotly=v5.19.0, polars=0.20.10

The code is put in [this repo](https://github.com/hopezh/test_quarto_dashboard), which was running correctly last night on another computer: 

````qmd
---
title: "test quarto dashboard"
format: dashboard
---

```{python}
import polars as pl
import plotly.express as px
import plotly.graph_objects as go
```

```{python}
# create df for solar farms
data_solar_farms = {
    "name"          : ["Setouchi Kirei", "Eurus Rokkasho", "Tomatoh Abira"],
    "capacity_mw"   : [235, 148, 111],
    "latitude"      : [34.6, 40.9, 42.7],
    "longitude"     : [134.1, 141.3, 141.7],
    "state"         : ["Okayama", "Aomori", "Hokkaido"],
    "city"          : ["Setouchi", "Rokkasho", "Abira"],
    "year_operation": [2018, 2015, 2015],
}
df_solar_farms = pl.DataFrame(data_solar_farms)
```

```{python}
# create df for typhoon paths
data_typhoon = {
    'name'            : ['Lan',  'Lan',  'Lan',  'Guc',  'Guc',  'Guc'],
    'idx'             : [230701, 230702, 230703, 220301, 220302, 220303],
    'latitude'        : [31.1,   36.6,   41.0,   31.8,   36.6,   37.8],
    'longitude'       : [137.3,  133.0,  135.8,  128.4,  135.0,  142.2],
    'date_time'       : [
        '2022-08-07 00:00:00',
        '2022-08-07 06:00:00',
        '2022-08-07 12:00:00',
        '2023-06-06 18:00:00',
        '2023-06-07 00:00:00',
        '2023-06-07 06:00:00',
    ],
    'grade'           : [1, 3, 2, 4, 6, 5]
}
df_typhoon = pl.DataFrame(data_typhoon)
```

## Row {height=70%}

### Column {width=50%}

```{python}
#| title: map solar farms
fig_solar_farms = go.Figure(
    go.Scattermapbox(
        lat = df_solar_farms['latitude'],
        lon = df_solar_farms['longitude'],
        mode = 'markers',
        marker = go.scattermapbox.Marker(
            size=df_solar_farms['capacity_mw']/10
        ),
    )
)

fig_solar_farms.update_layout(
    mapbox_style="carto-darkmatter",
    mapbox=dict(
        center=go.layout.mapbox.Center(
            lat=36,
            lon=138
        ),
        zoom=3
    ),
)
```

### Column {width=50%}

```{python}
#| title: map of typhoon paths
fig_typhoon = px.line_mapbox(
    df_typhoon,
    lat='latitude',
    lon='longitude',
    color='name',
    animation_frame='name',
)

fig_typhoon.update_layout(
    mapbox_style="carto-darkmatter",
    mapbox_center_lat=36,
    mapbox_center_lon=138,
    mapbox_zoom=3,
)
```

## Row {height=30%}
````

It's supposed to look like this:
![2024-02-22 13 28 11](https://github.com/quarto-dev/quarto-cli/assets/23435810/6281cd54-7f8d-46ba-99a7-f9256cf90bd4)

The output of my `quarto check`:
```txt
Quarto 1.4.550
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.550
      Path: C:\Users\jizha\AppData\Local\Programs\Quarto\bin
      CodePage: 936

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Tex:  (not detected)

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.7 (Conda)
      Path: C:/Users/jizha/.conda/envs/ai/python.exe
      Jupyter: 5.7.1
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.

Could you also ensure your example is small and does not contain unnecessary code? for example, is the commented code required to reproduce?

Currently, your code is not usable as you added backslash everywhere.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by hopezh** [reply]
Noted with thanks, @mcanouil. I've updated the code blocks.

**Comment by cderv** [comment, marked as answer]
I believe this is an issue, specific to Windows 

* https://github.com/quarto-dev/quarto-cli/issues/8857

You can follow it for a fix.

**Reply by cderv** [reply]
This will be fixed in next pre-release 1.5, but also backported in our next 1.4 patch release.

**Comment by ghirardinicola** [comment]
I got the same error with jupyter source, on mac

**Reply by mcanouil** [reply]
> I got the same error with jupyter source, on mac

Unfortunately, this does not help much.
Did you get the same error using the same code as initially shared? Another code?
Under 1.5 pre-release? 1.4.551?

**Comment by Zulfikar-TLL** [comment]
I get the same error with the same code but in `.ipynb` . `.qmd` format works fine.
Set up:
```
❯ quarto --version
1.5.57
❯ lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy
❯ code --version (vscode)
1.94.2
```

**Reply by cderv** [reply]
Can you share your ipynb file ? I did `quarto convert` on the .qmd example here, and then `quarto render doc.ipynb --execute` and I don't see the issue; 

So please, work on a new reproducible example with ipynb, test it with 1.5.57 and latest 1.6 pre-release. If you have issue, open a new one so that we can have a look. Thank you !

**Reply by Zulfikar-TLL** [reply]
@cderv Thanks for your reply. I also can't reproduce consistently. I will open a new one if I understand further.

**Reply by robmcd** [reply]
I too am getting the same error:

command:
`quarto render .\dira\dirb\dirc\dird\plotly_test.ipynb`

[plotly_test.zip (ipynb file)](https://github.com/user-attachments/files/19007987/plotly_test.zip)
[localhost-1740655029430.log](https://github.com/user-attachments/files/19008006/localhost-1740655029430.log)

If I use `embed-resources:true` in the header then I get a slightly different ERR_FILE_NOT_FOUND: [embed_resources-1740655745824.log](https://github.com/user-attachments/files/19008047/embed_resources-1740655745824.log)

plotly version: 5.24.1
Windows 10

quarto check:
```
Quarto 1.6.40
[>] Checking environment information...
      Quarto cache location: C:\Users\XXXXX\AppData\Local\quarto
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.4.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.46.3: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.6.40
      Path: C:\dev\apps\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Tex:  (not detected)

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.10.16
      Path: C:/xxxxxx/.venv/Scripts/python.exe
      Jupyter: 5.7.2
      Kernels: python3, glue_pyspark, glue_spark

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/

```
